From 0932d58eb26e53f9b3626d66120cbfe8cd7a8227 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Fri, 06 Aug 2021 09:28:05 +0300
Subject: [PATCH] [SW-50869] ppboot: use platform specific reset sources

Gaudi1 and Greco has common bit definitions in PSOC_GLOBAL_CONF_RST_SRC
register for reset cause. But in Gaudi2 reset cause bit definitions are
modified and not backward compatible.

updated usage of reset causes in multiple functions to do pcie
initialization only for COLD and PRSTN resets. Not assuming that rst_src
less than HL_WD_RST is only COLD and PRSTN resets.

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I04f9fdfceabfb7c4588515d35634ba27007091d9
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 1c22e89..c8f857c 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -114,21 +114,17 @@
 	case HL_COLD_RST:
 	case HL_PRSTN_RST:
 		/* PCIe is reset, so must initialize PCIe */
+		preboot_init_pcie();
 		break;
-	case HL_MANUAL_RST:
-	case HL_WD_RST:
-	case HL_SW_ALL_RST:
+	default:
 		/*
 		 * Currently, keep using the same flow as done before,
 		 * i.e. PCIe should already be initialized by default.
-		 * During hard reset triggered by LKD keep PCIe intact.
+		 * Except COLD & PRSTN resets, keep PCIe intact.
 		 */
-		return;
-	default:
 		break;
 	}
 
-	preboot_init_pcie();
 }
 
 static void rst_src_dump(void)
@@ -167,7 +163,7 @@
 
 	/* check source of reset */
 	rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
-	if (rst_src < HL_WD_RST)
+	if ((rst_src == HL_COLD_RST) || (rst_src == HL_PRSTN_RST))
 		return false;
 
 	prstn = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN);
diff --git a/lib/preboot/module_pci.c b/lib/preboot/module_pci.c
index 1ecfaa4..2a18d88 100644
--- a/lib/preboot/module_pci.c
+++ b/lib/preboot/module_pci.c
@@ -163,7 +163,8 @@
 	if (pci_sts.mem_enabled_prev != pci_sts.mem_enabled) {
 		pci_sts.mem_enabled_prev = pci_sts.mem_enabled;
 		rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
-		if (pci_sts.mem_enabled && (rst_src < HL_WD_RST))
+		if (pci_sts.mem_enabled && ((rst_src == HL_COLD_RST) ||
+					    (rst_src == HL_PRSTN_RST)))
 			hl_pci_iatu_init();
 
 		preboot_puts_raw("PCIe mem_en changed to: ");
